exporter/awsemfexporter/internal/appsignals/useragent.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package appsignals // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/internal/appsignals"
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/jellydator/ttlcache/v3"
+	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
+)
+
+const (
+	handlerName = "aws.appsignals.UserAgentHandler"
+	// defaultTTL is how long an item in the cache will remain if it has not been re-seen.
+	defaultTTL = time.Minute
+	// cacheSize is the maximum number of unique telemetry SDK languages that can be stored before one will be evicted.
+	cacheSize = 5
+	// attrLengthLimit is the maximum length of the language and version that will be used for the user agent.
+	attrLengthLimit = 20
+
+	// TODO: Available in semconv/v1.21.0+. Replace after collector dependency is v0.91.0+.
+	attributeTelemetryDistroVersion = "telemetry.distro.version"
+)
+
+type UserAgent struct {
+	mu          sync.RWMutex
+	prebuiltStr string
+	cache       *ttlcache.Cache[string, string]
+}
+
+func NewUserAgent() *UserAgent {
+	return newUserAgent(defaultTTL)
+}
+
+func newUserAgent(ttl time.Duration) *UserAgent {
+	ua := &UserAgent{
+		cache: ttlcache.New[string, string](
+			ttlcache.WithTTL[string, string](ttl),
+			ttlcache.WithCapacity[string, string](cacheSize),
+		),
+	}
+	ua.cache.OnEviction(func(context.Context, ttlcache.EvictionReason, *ttlcache.Item[string, string]) {
+		ua.build()
+	})
+	go ua.cache.Start()
```

Comment:
This is going to be a leaked goroutine, as it will block until `Stop` is called. [Docs reference](https://pkg.go.dev/github.com/jellydator/ttlcache/v3#Cache.Start).  The exporter's shutdown function should make sure `Stop` is called on the TTL cache.